ktor-client/ktor-client-cio/jvm/test/io/ktor/client/engine/cio/CIORequestTest.kt

```diff
@@ -45,4 +53,25 @@ class CIORequestTest : TestWithKtor() {
             }
         }
     }
+
+    @Test
+    fun testHangingTimeoutWithWrongUrl() = testWithEngine(CIO) {
+        config {
+            engine {
+                endpoint {
+                    connectTimeout = 1
+                }
+            }
+        }
+
+        test { client ->
+            for (i in 0..1000) {
+                try {
+                    client.get<String>("http://something.wrong")
```

Comment:
Is it good to make so much resolve requests, especially on TC? Don't we risk to be rate-limited here?

Reply:
It's the only way to reproduce the initial issue. I think it's ok